### PR TITLE
Fix #397: Crypto 3.2: Time synchronization

### LINKED ISSUE #397: Crypto 3.2: Time synchronization
Add information about server time for time synchronization between client and server.

- Add a new status `/pa/v3/status` endpoint which will contain information about server time.
- Return server time in `EciesEncryptedResponse`.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/ActivationController.java [hunk 1]
```diff
@@ -141,7 +141,9 @@ public ObjectResponse<ActivationRemoveResponse> removeActivation(
             logger.debug("Signature validation failed");
             throw new PowerAuthSignatureInvalidException();
         }
-        if (!"3.0".equals(apiAuthentication.getVersion()) && !"3.1".equals(apiAuthentication.getVersion())) {
+        if (!"3.0".equals(apiAuthentication.getVersion())
```
Comment (petrdvorak) [inline powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/ActivationController.java:144]: Just a note: This requires a note in the compatibility table (the latest SDKs would require this server version).

**Review by petrdvorak — APPROVED** [review]
Looks OK. There are two minor comments - one is that we likely do not use timestamp values from the new ECIES request/response in any way (in the scope of this PR), and one is about the mobile SDK docs compatibility table.

File: powerauth-restful-model/src/main/java/io/getlime/security/powerauth/rest/api/model/request/EciesEncryptedRequest.java [hunk 2]
```diff
@@ -30,6 +32,14 @@ public class EciesEncryptedRequest {
     private String encryptedData;
     private String mac;
     private String nonce;
+    private final long timestamp;
+
+    /**
+     * Request class constructor.
+     */
+    public EciesEncryptedRequest() {
+        this.timestamp = new Date().getTime();
```
[the remark this answers is not in the document]
Comment (romanstrobl) [reply powerauth-restful-model/src/main/java/io/getlime/security/powerauth/rest/api/model/request/EciesEncryptedRequest.java]: Fixed.

File: powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/ActivationController.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/ActivationController.java:144]: I will add a note into migration guide and we will update documentation wherever applicable.

**Review by hvge — APPROVED** [review]
Looks OK

File: powerauth-restful-model/src/main/java/io/getlime/security/powerauth/rest/api/model/request/EciesEncryptedRequest.java [hunk 3]
```diff
@@ -19,6 +19,8 @@
  */
 package io.getlime.security.powerauth.rest.api.model.request;
 
+import java.util.Date;
```
Comment (banterCZ) [inline powerauth-restful-model/src/main/java/io/getlime/security/powerauth/rest/api/model/request/EciesEncryptedRequest.java:22]: Unused import.

**Review by banterCZ — APPROVED** [review]

**Pull request merged**